src/Util/Text.cpp

```diff
@@ -23,19 +24,23 @@ Text::Text(const std::string &font, int fontSize, const std::string &text,
         InitUniformBuffer();
     }
 
+    auto surface =
+        std::unique_ptr<SDL_Surface, std::function<void(SDL_Surface *)>>();
     m_Font = {TTF_OpenFont(font.c_str(), fontSize), TTF_CloseFont};
 
     if (m_Font == nullptr) {
-        LOG_ERROR("Failed to load font: '{}'", text);
+        surface = {GetMissingFontTextureSDLSurface(), SDL_FreeSurface};
```

Comment:
Even if your functions won't throw, I think it would be better to log the error first.

Reply:
Hey I already do it :(